src/gtc/passes/oir_pipeline.py

```diff
@@ -70,11 +67,10 @@ def __init__(self, *, skip: Optional[Sequence[PassT]] = None):
     @staticmethod
     def all_steps() -> Sequence[PassT]:
         return [
-            graph_merge_horizontal_executions,
             AdjacentLoopMerging,
+            OnTheFlyMerging,
             LocalTemporariesToScalars,
             WriteBeforeReadTemporariesToScalars,
-            OnTheFlyMerging,
```

Comment:
Performing adjacent loop merging makes sense before the OnTheFlyMerging because that pass only looks within each `VerticalLoopSection`. Then, more temporaries can be made into scalars once horizontal executions are merged.